gnd/src/main/java/com/google/android/gnd/workers/TileDownloadWorker.java

```diff
@@ -42,106 +45,106 @@
 
   /**
    * Given a tile, downloads the given {@param tile}'s source file and saves it to the device's app
    * storage. Optional HTTP request header {@param requestProperties} may be provided.
    */
-  private Result downloadTileFile(Tile tile, Optional<HashMap<String, String>> requestProperties) {
-    try {
-      URL url = new URL(tile.getUrl());
-      HttpURLConnection connection = (HttpURLConnection) url.openConnection();
-
-      if (requestProperties.isPresent()) {
-        for (Map.Entry<String, String> property : requestProperties.get().entrySet()) {
-          connection.setRequestProperty(property.getKey(), property.getValue());
-        }
-      }
-
-      connection.connect();
-
-      InputStream is = connection.getInputStream();
-      FileOutputStream fos = context.openFileOutput(tile.getPath(), Context.MODE_PRIVATE);
-      byte[] byteChunk = new byte[BUFFER_SIZE];
-      int n;
-
-      while ((n = is.read(byteChunk)) > 0) {
-        fos.write(byteChunk, 0, n);
+  private void downloadTileFile(Tile tile, Map<String, String> requestProperties)
+      throws IOException {
+    URL url = new URL(tile.getUrl());
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+    if (!requestProperties.isEmpty()) {
+      for (Map.Entry<String, String> property : requestProperties.entrySet()) {
+        connection.setRequestProperty(property.getKey(), property.getValue());
       }
+    }
 
-      is.close();
-      fos.close();
-
-      localDataStore
-          .insertOrUpdateTile(tile.toBuilder().setState(Tile.State.DOWNLOADED).build())
-          .blockingAwait();
-
-      return Result.success();
-
-    } catch (IOException e) {
-      Log.d(TAG, "Failed to download and write file.", e);
+    connection.connect();
 
-      localDataStore
-          .insertOrUpdateTile(tile.toBuilder().setState(Tile.State.FAILED).build())
-          .blockingAwait();
+    InputStream is = connection.getInputStream();
+    FileOutputStream fos = context.openFileOutput(tile.getPath(), Context.MODE_PRIVATE);
+    byte[] byteChunk = new byte[BUFFER_SIZE];
+    int n;
 
-      return Result.failure();
+    while ((n = is.read(byteChunk)) > 0) {
+      fos.write(byteChunk, 0, n);
     }
-  }
-
-  /** Update a tile's state in the database and initiate a download of the tile source file. */
-  private Result downloadTile(Tile tile) {
-    localDataStore
-        .insertOrUpdateTile(tile.toBuilder().setState(Tile.State.IN_PROGRESS).build())
-        .blockingAwait();
 
-    return downloadTileFile(tile, Optional.empty());
+    is.close();
+    fos.close();
   }
 
-  /** Resumes downloading the source for {@param tile} marked as {@code Tile.State.IN_PROGRESS}. */
-  private Result resumeTileDownload(Tile tile) {
-    File existingTileFile = new File(context.getFilesDir(), tile.getPath());
+  /** Update a tile's state in the database and initiate a download of the tile source file. */
+  private Completable downloadTile(Tile tile) {
```

Comment:
Do I understand correct that "Tile" actually refers to an entire pyramid of tiles, and not a single tile? If so, at some point we may want to rename it something more appropriate to avoid confusing others (and ourselves).

Reply:
Have any ideas on what we should call it?

Reply:
Will probably hold off on this for a later PR

Reply:
SGTM. Filed #400 to track!